lua/nvim-tree/actions/search-node.lua

```diff
@@ -0,0 +1,39 @@
+local utils = require'nvim-tree.utils'
+local git = require"nvim-tree.git"
+local renderer = require'nvim-tree.renderer'
+
+local M = {
+}
+
+function M.fn()
+  if not TreeExplorer then return end
+
+  local input_path = vim.fn.input("Search node: ", "", "file")
+  utils.clear_prompt()
+
+  absolute_input_path = TreeExplorer.cwd .. utils.path_separator .. input_path
+
+  local function search_node(nodes)
+    for _, node in ipairs(nodes) do
+      matches = utils.str_find(absolute_input_path, node.absolute_path .. utils.path_separator)
+      if matches then
+        if node.nodes then
+          if not node.open then
+            node.open = true
+            TreeExplorer:expand(node)
+            renderer.draw()
```

Comment:
expand already does `renderer.draw` once it is finished. But actually, it's not a good idea to use this yet because it's asynchronous (i have to make it synchronous). Thus the following search_node could give an empty table.